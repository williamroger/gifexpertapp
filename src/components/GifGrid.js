import React, { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category).then(imgs => setImages(imgs));
  }, [category]);


  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map(image => (
          <GifGridItem 
            key={image.id}
            {...image}
          />
        ))}
      </div>
    </>
  );
};
