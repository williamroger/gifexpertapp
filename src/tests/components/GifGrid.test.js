import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Tests for component GifGrid', () => {
  const category = 'Spiderman';

  test('should be render GifGrid', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should be render GifGrid with imagens of useFetchGifs', () => {
    const gifs = [{
      id: '123ABC',
      title: 'Poster Spiderman',
      url: 'http://marvel.com/poster/spiderman.jpg',
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});