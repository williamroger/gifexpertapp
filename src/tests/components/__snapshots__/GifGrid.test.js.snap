// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for component GifGrid should be render GifGrid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Spiderman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests for component GifGrid should be render GifGrid with imagens of useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Spiderman
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123ABC"
      key="123ABC"
      title="Poster Spiderman"
      url="http://marvel.com/poster/spiderman.jpg"
    />
  </div>
</Fragment>
`;
